app/controllers/course/lesson_plan/items_controller.rb

```diff
@@ -67,10 +69,17 @@ def tab_title_array(tab)
     tab.category.tabs.size > 1 ? [category_name, tab.title] : [category_name]
   end
 
-  # Gets the array of actable type names for lesson plan items of enabled components.
+  # Gets a hash of actable type names for lesson plan items of enabled components mapped to data
+  # that will be passed to actable's model scope for further processing.
   #
-  # @return [Array<String>] Array of actable_type names.
-  def enabled_lesson_plan_item_actable_type_names
-    current_component_host.enabled_components.map(&:lesson_plan_item_actable_names).flatten
+  # @return [Hash{String} => Array or nil] Hash of actable_type names.
+  def actable_hash
+    @item_settings.actable_hash(current_component_host.enabled_components.
+                                map(&:lesson_plan_item_actable_names).flatten)
```

Comment:
Is it cleaner to let component instances delegate calls to `lesson_plan_item_actable_names` to their classes (like [this](https://github.com/Coursemology/coursemology2/blob/master/app/controllers/components/course/controller_component_host.rb#L44-L46))? 

Can do something like
```
class Course::Settings::LessonPlanItems
  def enabled_actable_types
    @components.map(&:lesson_plan_item_actable_names).flatten
  end
end
```
and
```
def index
  @items = @items.with_actable_types(@item_settings.enabled_actable_types)
  ....
end
```